frontend/packages/console-shared/src/components/dashboard/DashboardGrid.tsx

```diff
@@ -22,40 +22,53 @@ const mapCardsToGrid = (
     </GridItem>
   ));
 
-const DashboardGrid: React.FC<DashboardGridProps> = ({
-  mainCards,
-  leftCards = [],
-  rightCards = [],
-}) => {
+const DashboardGrid: React.FC<DashboardGridProps> = ({ mainCards, leftCards, rightCards }) => {
   const [containerRef, width] = useRefWidth();
-  const grid =
-    width <= parseInt(breakpointLG.value, 10) ? (
-      <Grid className="co-dashboard-grid">
-        <GridItem lg={12} md={12} sm={12}>
-          <Grid className="co-dashboard-grid">{mapCardsToGrid(mainCards, 'main', true)}</Grid>
-        </GridItem>
-        <GridItem key="left" lg={12} md={12} sm={12}>
-          <Grid className="co-dashboard-grid">{mapCardsToGrid(leftCards, 'left', true)}</Grid>
-        </GridItem>
-        <GridItem key="right" lg={12} md={12} sm={12}>
-          <Grid className="co-dashboard-grid">{mapCardsToGrid(rightCards, 'right', true)}</Grid>
-        </GridItem>
-      </Grid>
-    ) : (
-      <Grid className="co-dashboard-grid">
-        <GridItem key="left" lg={3} md={3} sm={3}>
-          <Grid className="co-dashboard-grid">{mapCardsToGrid(leftCards, 'left')}</Grid>
-        </GridItem>
-        <GridItem lg={6} md={6} sm={6}>
-          <Grid className="co-dashboard-grid">{mapCardsToGrid(mainCards, 'main')}</Grid>
-        </GridItem>
-        <GridItem key="right" lg={3} md={3} sm={3}>
-          <Grid className="co-dashboard-grid">{mapCardsToGrid(rightCards, 'right')}</Grid>
-        </GridItem>
-      </Grid>
-    );
+  const smallGrid =
+    !!containerRef.current && width !== null && width <= parseInt(breakpointLG.value, 10);
```

Comment:
Just a minor suggestion:
May be we can use optional chaining here : https://github.com/openshift/console/blob/229af0a99a87218410a6f4704c89cbb5efa86646/frontend/public/components/utils/ref-width-hook.ts#L7-L10

Then ` width !== null `check can be removed.

Reply:
fixed